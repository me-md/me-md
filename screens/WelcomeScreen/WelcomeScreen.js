import React from 'react';
import { Dimensions, Image, StyleSheet, View } from 'react-native';
import { Button, Text } from 'native-base';
import { Header } from '../../components/Header';

const height = Dimensions.get('window').height;

export default function WelcomeScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <Header />
      <Image
        source={require('../../assets/images/doctor.png')}
        style={styles.icon}
      />
      <View style={styles.welcome}>
        <Text style={styles.greeting}>Hello!</Text>
        <Text style={styles.message}>
          Welcome to MeMD, an online platform designed to help diagnose symptoms
          and locate potential doctors. Please click BEGIN to start your health
          checkup.
        </Text>
        <Button
          block
          style={styles.button}
          onPress={() => navigation.navigate('TermsAndConditions')}
        >
          <Text>BEGIN CHECKUP</Text>
        </Button>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFFFFF',
    flex: 1
  },
  icon: {
    alignSelf: 'center',
    borderWidth: 0,
    borderRadius: 50,
    height: height * 0.2,
    marginBottom: height * 0.01,
    marginTop: height * 0.04,
    width: height * 0.2
  },
  greeting: {
    fontSize: 40,
    fontWeight: 'bold',
    margin: height * 0.03
  },
  message: {
    fontSize: 26,
    marginLeft: height * 0.03,
    marginBottom: height * 0.03
  },
  button: {
    alignSelf: 'center',
    marginTop: height * 0.02,
    width: '75%'
  }
});
