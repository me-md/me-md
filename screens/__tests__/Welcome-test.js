import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import WelcomeScreen from '../../screens/WelcomeScreen/WelcomeScreen';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

describe('WelcomeScreen', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`should match a snapshot`, () => {
    const tree = renderer.create(<WelcomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
