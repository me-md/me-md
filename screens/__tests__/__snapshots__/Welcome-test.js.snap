// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should match a snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingBottom": 100.05,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278210303,
          4287155441,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "height": 200.1,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 50,
        "borderWidth": 0,
        "height": 266.8,
        "marginBottom": 13.34,
        "marginTop": 53.36,
        "width": 266.8,
      }
    }
  />
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "fontSize": 40,
            "fontWeight": "bold",
            "margin": 40.019999999999996,
          },
        ]
      }
      uppercase={false}
    >
      Hello!
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "fontSize": 26,
            "marginBottom": 40.019999999999996,
            "marginLeft": 40.019999999999996,
          },
        ]
      }
      uppercase={false}
    >
      Welcome to MeMD, an online platform designed to help diagnose conditions and locate potential doctors. Please click BEGIN CHECKUP to start your health checkup.
    </Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginTop": 26.68,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "width": "75%",
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        BEGIN CHECKUP
      </Text>
    </View>
  </View>
</View>
`;
